### PR TITLE
Implement 7 ULDUM cards and ability 'Lifesteal'

### PR DESCRIPTION
**Pull request by 1cekrim**
This revision includes:  
- Implement 7 ULDUM cards
    - Beaming Sidekick (ULD_191)
    - Bug Collector (ULD_712)
    - Serpent Egg (ULD_174)
    - Injured Tol'vir (ULD_271)
    - Vilefiend (ULD_450)
    - Spitting Camel (ULD_182)
    - Kobold Sandtrooper (ULD_184)
- Implement ability `Lifesteal`

### CODE REVIEW

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Tests/UnitTests/CardSets/UldumCardsGenTests.cpp  225
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4928493&bid=16303882)

File: Sources/Rosetta/CardSets/UldumCardsGen.cpp [hunk 1]
```diff
@@ -2129,10 +2171,15 @@ void UldumCardsGen::AddNeutralNonCollect(PowersType& powers,
...
 {
+    Power power;
+
     // --------------------------------------- MINION - NEUTRAL
     // [ULD_174t] Sea Serpent (*) - COST:3 [ATK:3/HP:4]
     // - Race: Beast, Set: Uldum
     // --------------------------------------------------------
+    power.ClearData();
+    power.AddPowerTask(nullptr);
+    powers.emplace("ULD174t", power);
```
Comment (utilForever) [inline Sources/Rosetta/CardSets/UldumCardsGen.cpp]: Errata: `ULD174t` -> `ULD_174t`

File: Tests/UnitTests/CardSets/UldumCardsGenTests.cpp [hunk 2]
```diff
@@ -6,7 +6,388 @@
...
+
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+    EXPECT_EQ(curField.GetCount(), 1);
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_START);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card2, card1));
+    EXPECT_EQ(curField.GetCount(), 1);
+    EXPECT_EQ(curField[0]->GetAttack(), 3);
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/UldumCardsGenTests.cpp]: 죽음의 메아리를 통해 소환된 하수인의 이름과 체력도 확인하는 코드가 있으면 좋겠네요.

File: Tests/UnitTests/CardSets/UldumCardsGenTests.cpp [hunk 3]
```diff
@@ -6,7 +6,388 @@
...
+    auto& curField = *(curPlayer->GetFieldZone());
+    auto& opField = *(opPlayer->GetFieldZone());
+
+    const auto card1 =
+        Generic::DrawCard(curPlayer, Cards::FindCardByName("Bug Collector"));
+    const auto card2 =
+        Generic::DrawCard(opPlayer, Cards::FindCardByName("Bug Collector"));
+
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+    EXPECT_EQ(curField.GetCount(), 2);
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/UldumCardsGenTests.cpp]: 전투의 함성을 통해 소환된 하수인의 이름/공격력/체력/속공 여부를 확인하는 코드가 추가되면 좋겠네요.

File: Sources/Rosetta/CardSets/UldumCardsGen.cpp (hunk 1 above)
Comment (1cekrim) [reply Sources/Rosetta/CardSets/UldumCardsGen.cpp]: 수정했습니다.

File: Tests/UnitTests/CardSets/UldumCardsGenTests.cpp (hunk 2 above)
Comment (1cekrim) [reply Tests/UnitTests/CardSets/UldumCardsGenTests.cpp]: 수정했습니다.

File: Tests/UnitTests/CardSets/UldumCardsGenTests.cpp (hunk 3 above)
Comment (1cekrim) [reply Tests/UnitTests/CardSets/UldumCardsGenTests.cpp]: 수정했습니다.

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Tests/UnitTests/CardSets/UldumCardsGenTests.cpp  225
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4928493&bid=16303882)

**Pull request merged**